Flusher: Panic on startup

#### Describe the bug
Running the `flusher` panics:

```
level=info ts=2022-08-31T06:57:58.354300237Z caller=main.go:235 msg="starting Grafana Enterprise Metrics" version="(version=r201-6b5ebb93, branch=weekly-r201, revision=6b5ebb93)"
level=info ts=2022-08-31T06:57:58.365796748Z caller=added_modules.go:177 msg="management object storage client not starting, no backend configured"
level=info ts=2022-08-31T06:57:58.39454334Z caller=quota.go:297 msg="registered quota collector"
level=warn ts=2022-08-31T06:57:58.394659298Z caller=manager.go:169 msg="unable to load local license" error="license token file not found: ./license.jwt"
level=info ts=2022-08-31T06:57:58.396944689Z caller=filesystem.go:46 msg="excluding mount points" mounts=^/(dev|proc|sys|var/lib/docker/.+)($|/)
level=info ts=2022-08-31T06:57:58.397011638Z caller=filesystem.go:49 msg="excluding filesystem types" types=^(autofs|binfmt_misc|bpf|cgroup2?|configfs|debugfs|devpts|devtmpfs|fusectl|hugetlbfs|iso9660|mqueue|nsfs|overlay|proc|procfs|pstore|rpc_pipefs|securityfs|selinuxfs|squashfs|sysfs|tracefs)$
level=info ts=2022-08-31T06:57:58.397799301Z caller=server.go:306 http=[::]:80 grpc=[::]:9095 msg="server listening on addresses"
level=info ts=2022-08-31T06:57:58.442418315Z caller=module_service.go:82 msg=initialising module=activity-tracker
level=info ts=2022-08-31T06:57:58.442455851Z caller=module_service.go:82 msg=initialising module=distributor-bi-push-wrapper
level=info ts=2022-08-31T06:57:58.442468525Z caller=module_service.go:82 msg=initialising module=license-manager
level=info ts=2022-08-31T06:57:58.442472355Z caller=module_service.go:82 msg=initialising module=sanity-check
level=info ts=2022-08-31T06:57:58.442486336Z caller=manager.go:132 msg="license manager up and running"
level=info ts=2022-08-31T06:57:58.442524065Z caller=sanity_check.go:32 msg="Checking directories read/write access"
level=info ts=2022-08-31T06:57:58.442540164Z caller=sanity_check.go:37 msg="Directories read/write access successfully checked"
level=info ts=2022-08-31T06:57:58.442548902Z caller=sanity_check.go:39 msg="Checking object storage config"
level=info ts=2022-08-31T06:57:58.442561533Z caller=sanity_check.go:44 msg="Object storage config successfully checked"
level=info ts=2022-08-31T06:57:58.44259187Z caller=module_service.go:82 msg=initialising module=server
level=info ts=2022-08-31T06:57:58.442719863Z caller=module_service.go:82 msg=initialising module=flusher
level=info ts=2022-08-31T06:57:58.44282715Z caller=mimir.go:758 msg="Application started"
level=info ts=2022-08-31T06:57:58.446577124Z caller=ingester.go:1659 msg="opening existing TSDBs"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x178d039]

goroutine 280 [running]:
github.com/grafana/mimir/pkg/util/validation.(*Overrides).ActiveSeriesCustomTrackersConfig(0x2ce6e20?, {0xc00084b0fb?, 0xc000c08e50?})
	/backend-enterprise/vendor/github.com/grafana/mimir/pkg/util/validation/limits.go:522 +0x19
github.com/grafana/mimir/pkg/ingester.(*Ingester).createTSDB(0xc000b3c000, {0xc00084b0fb, 0x5})
	/backend-enterprise/vendor/github.com/grafana/mimir/pkg/ingester/ingester.go:1531 +0x349
github.com/grafana/mimir/pkg/ingester.(*Ingester).openExistingTSDB.func1()
	/backend-enterprise/vendor/github.com/grafana/mimir/pkg/ingester/ingester.go:1670 +0x165
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/backend-enterprise/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/backend-enterprise/vendor/golang.org/x/sync/errgroup/errgroup.go:72 +0xa5
```

#### To Reproduce
Steps to reproduce the behavior:

1. Start Mimir (r201) with `-target=flusher` on an ingester disk _with data_

#### Expected behavior

It doesn't panic.

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: jsonnet

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@
 
 ### Grafana Mimir
 
+* [BUGFIX] Flusher: Add `Overrides` as a dependency to prevent panics when starting with `-target=flusher`. #3151
+
 ### Mixin
 
 ### Jsonnet

--- a/pkg/mimir/modules.go
+++ b/pkg/mimir/modules.go
@@ -831,7 +831,7 @@ func (t *Mimir) setupModuleManager() error {
 		DistributorService:       {Ring, Overrides},
 		Ingester:                 {IngesterService, API},
 		IngesterService:          {Overrides, RuntimeConfig, MemberlistKV},
-		Flusher:                  {API},
+		Flusher:                  {Overrides, API},
 		Queryable:                {Overrides, DistributorService, Ring, API, StoreQueryable, MemberlistKV},
 		Querier:                  {TenantFederation},
 		StoreQueryable:           {Overrides, MemberlistKV},

